roles/node/tasks/main.yml

```diff
@@ -0,0 +1,111 @@
+---
+- name: Include network UFW rules vars file
+  ansible.builtin.include_vars:
+    file: vars/ufw_rules.yml
+
+- name: Apply UFW rules
+  ansible.builtin.include_tasks: tasks/apply_ufw_rules.yml
+  loop: "{{ node_ufw_rules }}"
+
+- name: Include user creation and directories task
+  ansible.builtin.include_tasks: tasks/create_user_and_directories.yml
+  vars:
+    home: "{{ node_home_dir }}"
+    directories: "{{ node_directories }}"
+    log_file_path: "{{ node_log_file_path }}"
+
+- name: Install Cometbft
+  ansible.builtin.include_tasks: tasks/extract_binary.yml
+  vars:
+    arch_details: "{{ ansible_system | lower }}_{{ system_architecture }}"
+    build_suffix_url: "v{{ namada[type].cometbft.version }}/{{ node_cometbft_default_binary_name }}_{{ namada[type].cometbft.version }}"
+    source_url: "{{ cometbft.source_url }}"
+    suffix_url: "{{ build_suffix_url }}_{{ arch_details }}.{{ cometbft.file_extension }}"
+    dest_path: "{{ node_bin_dir }}"
+    extracted_path: 
+      - "{{ node_cometbft_default_binary_name }}"
+
+- name: Install Namada
+  ansible.builtin.include_tasks: tasks/extract_binary.yml
+  vars:
+    arch_details: "{{ ansible_system }}-{{ ansible_architecture }}"
+    build_suffix_url: "{{ namada[type].version }}/{{ node_service_default_binary_name[0] }}-{{ namada[type].version }}"
+    source_url: "{{ namada.source_url }}"
+    suffix_url: "{{ build_suffix_url }}-{{ arch_details }}.{{ namada.file_extension }}"
+    dest_path: "{{ node_bin_dir }}"
+    extracted_dir: "{{ node_service_default_binary_name[0] }}-{{ namada[type].version }}-{{ arch_details }}"
+    extracted_path:
```

Comment:
1. is there a better way to do it ? maybe loop over the list in one go ? or create a map with the default names and the name we want ?
2. atleast one of these must be explicitly renamed to service identifier, or it wont work if its anything else than default node type